### PR TITLE
Fixed relevant issues to match ControlNetVersioner unit tests to truthdata.

### PR DESCRIPTION
**Pull request by AgoinsUSGS**
(1): Added HasChooserName() and HasDateTime() methods to ControlPoint and ControlMeasure to allow us to check if these values are empty without overriding them. (Using the GetChooserName() and GetDateTime() would override these values if they were empty).

(2): Added checks for Apriori Lat/Long values to ensure that they're not Null before we write them. These values were being written previously because we were only checking if aprioriCovarianceMatrix.size1() > 0 before writing the matrix.

(3): In ControlPointV0001, fixed an issue where we were seeting ControlPoint::Ignore to true where we should be setting the measure to true.

(4): In ControlPointV0001, initialize double values to Isis::Null by default. Some values were reading in with a default of 0. 

(5): Updated ControlNetVersioner's unitTest to match truthdata spacing.

### CODE REVIEW

**Comment by AgoinsUSGS** [comment]
[tests failing - control net versioner](https://trello.com/c/6miDMVKf/158-tests-failing-control-net-versioner)

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 1]
```diff
@@ -914,7 +924,7 @@ namespace Isis {
       }
       catch (IException &e) {
         QString msg = "Failed to convert version 1 protobuf control point at index ["
-                      + toString(i) + "] into a ControlPoint.";
+                      + toString(i) + "] to a ControlPoint.";
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: I think this should still be "into."

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 2]
```diff
@@ -1134,7 +1144,7 @@ namespace Isis {
       }
       catch (IException &e) {
         QString msg = "Failed to convert protobuf version 2 control point at index ["
-                      + toString(pointIndex) + "] into a ControlPoint.";
+                      + toString(pointIndex) + "] in a ControlPoint.";
```
Comment (kberryUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: I think this should also still be "into"

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 3]
```diff
@@ -465,7 +473,8 @@ namespace Isis {
           pvlMeasure += PvlKeyword("Line", toString(controlMeasure.GetLine()));
         }
 
-        if ( controlMeasure.GetDiameter() != Isis::Null ) {
+        if ( controlMeasure.GetDiameter() != Isis::Null
```
Comment (AgoinsUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: The old truth data would omit a diameter if it's value was 0. That's why this additional check was implemented.

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 4]
```diff
@@ -465,7 +473,8 @@ namespace Isis {
...
+             && controlMeasure.GetDiameter() != 0. ) {
```
Comment (AgoinsUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: The old truth data would omit a diameter if it's value was 0. That's why this additional check was implemented.

File: isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp [hunk 5]
```diff
@@ -1761,7 +1771,11 @@ namespace Isis {
         protoPoint.set_aprioriz(aprioriSurfacePoint.GetZ().meters());
 
         symmetric_matrix<double, upper> aprioriCovarianceMatrix = aprioriSurfacePoint.GetRectangularMatrix();
-        if ( aprioriCovarianceMatrix.size1() > 0 ) {
+        if ( aprioriCovarianceMatrix.size1() > 0 &&
+             aprioriSurfacePoint.GetLatSigmaDistance().meters() != Isis::Null &&
+             aprioriSurfacePoint.GetLonSigmaDistance().meters() != Isis::Null &&
+             aprioriSurfacePoint.GetLocalRadiusSigma().meters() != Isis::Null) {
```
Comment (AgoinsUSGS) [inline isis/src/control/objs/ControlNetVersioner/ControlNetVersioner.cpp]: Previously we weren't checking that the Lat/Long sigmas had valid values before writing them out.

**Comment by AgoinsUSGS** [comment]
[SampleResidual and LineResidual are being set to 0.0 when writing a Pvl0001 as a Pvl0005](https://trello.com/c/bkgUz5iX/194-sampleresidual-and-lineresidual-are-being-set-to-00-when-writing-a-pvl0001-as-a-pvl0005)

**Pull request merged**